src/containers/blocks.jsx

```diff
@@ -55,9 +60,18 @@ class Blocks extends React.Component {
         this.attachVM();
     }
     shouldComponentUpdate (nextProps, nextState) {
-        return this.state.prompt !== nextState.prompt || this.props.isVisible !== nextProps.isVisible;
+        return (
+            this.state.prompt !== nextState.prompt ||
+            this.props.isVisible !== nextProps.isVisible ||
+            this.props.toolboxXML !== nextProps.toolboxXML
+        );
     }
     componentDidUpdate (prevProps) {
+        if (prevProps.toolboxXML !== this.props.toolboxXML) {
+            const selectedCategoryName = this.workspace.toolbox_.getSelectedItem().name_;
+            this.workspace.updateToolbox(this.props.toolboxXML);
+            this.setToolboxSelectedItemByName(selectedCategoryName);
```

Comment:
Can you add a comment around why this is needed? Is it a bug with scratch-blocks? If so, could you file an issue and leave a `@todo` here. Even if it isn't a bug, we should still leave a todo about modifying the scratch-blocks workspace/toolbox API so that we don't have to use properties that are marked as "private" (the trailing underscore).

Reply:
TBH, I just copied this from @ericrosenbaum's work. It's possible that it's no longer necessary. I mean, the `updateToolbox` is necessary but I can try without the category switching logic.